src/Expr.js

```diff
@@ -169,35 +236,109 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    var argumentsExpr = null
+
+    if (isExpr(expr['arguments'])) {
```

Comment:
this whole logic is not necessary, just calling `argumentsExpr = exprToString(expr['arguments'])` should be enough, all the logic added here was already handled in the beginning of the function

Reply:
I added this check because one of the pre-existing unit tests was getting to this point where the `arguments` and `call` values were both expressions 🤔